src/backend/distributed/executor/multi_utility.c

```diff
@@ -296,6 +296,35 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		return;
 	}
 
+#if (PG_VERSION_NUM >= 110000)
+	if (IsA(parsetree, CallStmt))
+	{
+		/*
+		 * Stored procedures are a bit strange in the sense that some statements
+		 * are not in a transaction block, but can be rolled back. We need to
+		 * make sure we send all statements in a transaction block. The
+		 * IsStoredProcedure flag signals this to the router executor.
+		 */
+		IsStoredProcedure = true;
+
+		PG_TRY();
+		{
+			standard_ProcessUtility(pstmt, queryString, context,
+									params, queryEnv, dest, completionTag);
+
+			IsStoredProcedure = false;
```

Comment:
we should consider a counter instead of a boolean flag here

Reply:
Otherwise our execution is incorrect when nested procedure calls are used

Reply:
Wow, I didn't realise procedure calls can be nested. That must get really confusing with commit/rollback. Will change to a counter.